.circleci/config.yml

```diff
@@ -598,10 +598,14 @@ jobs:
       - run:
           name: Lint DockerFile changes
           command: |
-            eval `.circleci/get_pr_info.sh -b`
-            CHANGED_DOCKER_FILES=`cat "$CHANGED_FILE_OUTPUTFILE" | grep Dockerfile | tr '\n' ' '`
-            if [[ ! -z "$CHANGED_DOCKER_FILES" ]]; then
-              hadolint -c .lintrules/hadolint.yaml $CHANGED_DOCKER_FILES || true
+            target_info=$(.circleci/get_pr_info.sh -b)
+            echo "$target_info"
+            eval "$target_info"
+            if  [[ -n "$BASE_GITHASH" ]] && [[ -n "$CHANGED_FILE_OUTPUTFILE" ]]; then
+              CHANGED_DOCKER_FILE_LIST=$( grep Dockerfile $CHANGED_FILE_OUTPUTFILE  || true | tr '\n' ' ' || true )
```

Comment:
apparently grep will exit 1 if it doesn't find matches....  found this to be very weird, and couldn't repro on my mac, but this works.

Reply:
TBH, piping true to another cmd then or-ing true feels weird.  

If `tr` fails, $CHANGED_DOCKER_FILE_LIST would become `"true"`.  I think the hadolint below will fail.

Reply:
true isn't getting piped -- looks like || is taking precedence, will add parens to make it clear.    the second || true probably isn't needed

Reply:
also it's grep that is failing if it doesn't find anything,  like so... https://stackoverflow.com/questions/42251386/linux-grep-return-code